### DISCUSSION TITLE
Incorrect typing

### DISCUSSION THREAD

**Discussion by HuHguZ** [sindresorhus/got#1816] [Q&A]
https://github.com/sindresorhus/got/blob/main/documentation/10-instances.md

How do I fix the typing differences between the real code and what the typescript tells me?

![image](https://user-images.githubusercontent.com/32592443/127748050-9df5086a-1191-443e-b25b-d47af4d3d025.png)
![image](https://user-images.githubusercontent.com/32592443/127748069-db23b04c-262f-48a4-997d-f843fdfe8373.png)

**Comment by HuHguZ** [comment]
![image](https://user-images.githubusercontent.com/32592443/127748120-aa3c9f08-9e18-49ed-80a7-1f45f680651f.png)

**Comment by szmarczak** [comment]
Cast `response` to [`Response`](https://github.com/sindresorhus/got/blob/846e298b6b48a7b78e46eb487b4fdc0ffc0f8cd9/source/core/response.ts#L95). In your code TypeScript doesn't know if this is a stream or a promise.

**Reply by HuHguZ** [reply]
![image](https://user-images.githubusercontent.com/32592443/127753333-dbea057e-fcf6-47ec-a8a5-6e1b1b2aa1a7.png)

**Reply by szmarczak** [reply]
You're trying to cast the promise. You need to cast the actual value instead.

**Reply by HuHguZ** [reply]
Same problem, can you show me an example of how to solve the problem?

**Reply by szmarczak** [reply]
`(await next(options)) as unknown as Response<unknown>` should do
